sourcegraph/sourcegraph
# Tree: add "go up" the tree button

## Description
Part of the [Tree improvements](https://github.com/sourcegraph/sourcegraph/issues/38667), it adds the option to [navigate up the tree](https://github.com/sourcegraph/sourcegraph/issues/38663).

The tree still retains the "confusing behaviour", which is the inconsistency between:

1. Navigating to the root of the repo - which opens up the entire tree at root, as expected
2. Going to a deeply nested file, which only opens the subtree - which is confusing

This part will be fixed later on, with a bigger refactoring of the entire `Tree` component, including data fetching, tree processing, and rendering.

## Test plan
1. Check out the branch
4. Open up a few repos:
    1. A regular one with lots of subfolders: https://sourcegraph.test:3443/github.com/sourcegraph/code-intel-extensions
    2. Something with a deeply nested single children: https://sourcegraph.test:3443/github.com/sgtest/java-pom-with-deps-javaconfig/-/blob/src/main/java/com/sourcegraph/depuser/ApacheCommonsUser.java
    5. A single-file one: https://sourcegraph.test:3443/github.com/sgtest/head-branch/-/blob/foo
3. Make sure that ".." is shown and navigates up the tree
6. Make sure that tests are passing


https://user-images.githubusercontent.com/2196347/180196529-e712b6cb-2fef-4185-8079-2b12a328b4fb.mov



## App preview:

- [Web](https://sg-web-og-tree-nav-up.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-yudpsyqufg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
